autolab/Autolab
# Add docker image uploading

## Summary

### Summary generated by Reviewpad on 29 Jan 24 23:11 UTC
This pull request includes the following changes:

1. Addition of a new file `dockers_controller.rb` in the `app/controllers` directory.
2. Modification in the existing file `assessment_autograde_core.rb` in the `app/helpers` directory.
3. Modification in the existing file `index.html.erb` in the `app/views/assessments` directory.
4. Modification in the existing file `_form.html.erb` in the `app/views/autograders` directory.
5. Addition of a new file `index.html.erb` in the `app/views/dockers` directory.
6. Modification in the existing file `development.rb` in the `config/environments` directory.
7. Modification in the existing file `production.rb.template` in the `config/environments` directory.
8. Modification in the existing file `test.rb` in the `config/environments` directory.
9. Modification in the existing file `routes.rb` in the `config` directory.
10. Modification in the existing file `tango_client.rb` in the `lib` directory.

Please refer to the diff for more details.


## Description
- Add page for uploading docker images.
- Add link to docker upload image on manage course page and edit autograder page.
- Corresponding [Tango PR](https://github.com/autolab/Tango/pull/232).
- Addresses: #2002.

## How Has This Been Tested?
With both local and docker installs:
1. Add an assessment that uses a docker image (created by `docker save`) not yet used by Tango.
2. Confirm that attempting to autograde a submission will cause an error due to the missing image.
3. Upload the docker image using the new upload docker image page. 
4. Confirm that autograding submissions on that assessment now works. 

## Types of changes
